container/factory.go

```diff
@@ -89,30 +86,27 @@ func HasFactories() bool {
 }
 
 // Create a new ContainerHandler for the specified container.
-func NewContainerHandler(name string, inHostNamespace bool) (ContainerHandler, bool, error) {
+func NewContainerHandler(name string, inHostNamespace bool) (ContainerHandler, error) {
 	factoriesLock.RLock()
 	defer factoriesLock.RUnlock()
 
 	// Create the ContainerHandler with the first factory that supports it.
 	for _, factory := range factories {
-		canHandle, canAccept, err := factory.CanHandleAndAccept(name)
+		handler, ignore, err := factory.NewContainerHandler(name, inHostNamespace)
```

Comment:
I'm worried that the ignore handler  will unnecessary complicate things, especially if the results of some of those functions are assumed to be non-nill if the error is nil.

> but doesn't this prevent NewContainerHandler from returning other errors that might occur?

It means we'd change the behaviour of an error, and say that errors are returned rather than logged & ignored (in NewContainerHandler). If the error is bubbled up to manager the behaviour is still consistent ([manager.go#L774](https://github.com/google/cadvisor/blob/5d7c71a49536d56729c7c778daa70653c138f4a5/manager/manager.go#L774)). Tracing up through the code, I think this is actually preferable. For example, consider the case of creating the root container ([manager.go#L261](https://github.com/google/cadvisor/blob/5d7c71a49536d56729c7c778daa70653c138f4a5/manager/manager.go#L261)): ignoring the container in that case _should_ be an actual error, and not silently ignored. Everywhere else it will simply be logged (e.g. "E: Failed to create existing container: foo: ignoring container foo").